src/textEditor.ts

```diff
@@ -160,6 +160,14 @@ export class TextEditor {
     return word;
   }
 
+  static getTabCharacter(): string {
```

Comment:
I know we handle `expandtab` in other places. Are you sure you aren't duplicating this logic? Or maybe we've done it ad-hoc in a few places and this should replace those.

Reply:
Thanks for the review. I'll check and get back shortly.

Reply:
After quick global search for `expandtab`, I found one method that I think could be refactored to use this `getTabCharacter()` method. But I don't know how, it's in `TextEditor.setIndentationLevel`.

```javascript
  static setIndentationLevel(line: string, screenCharacters: number): string {
    let tabSize = configuration.tabstop;
    let insertTabAsSpaces = configuration.expandtab;

    if (screenCharacters < 0) {
      screenCharacters = 0;
    }

    let indentString = '';

    if (insertTabAsSpaces) {
      indentString += new Array(screenCharacters + 1).join(' ');
    } else {
      if (screenCharacters / tabSize > 0) {
        indentString += new Array(Math.floor(screenCharacters / tabSize) + 1).join('\t');
      }

      indentString += new Array((screenCharacters % tabSize) + 1).join(' ');
    }

    let firstNonWhiteSpace = 0;
    let lineCheck = line.match(/^\s*/);
    if (lineCheck) {
      firstNonWhiteSpace = lineCheck[0].length;
    }

    return indentString + line.substring(firstNonWhiteSpace, line.length);
  }
```

Reply:
Okay, no worries - thanks for checking. We can always come back to this for a refactor.